tests/unit/server/bundle_manager/make_bundles_test.py

```diff
@@ -0,0 +1,272 @@
+from codalab.worker.bundle_state import Dependency, State
+from codalab.objects.dependency import Dependency
+from codalab.bundles.run_bundle import RunBundle
+from codalab.bundles.make_bundle import MakeBundle
+from codalab.bundles.dataset_bundle import DatasetBundle
+from codalab.lib.spec_util import generate_uuid
+import os
+import tempfile
+from tests.unit.server.bundle_manager import (
+    BaseBundleManagerTest,
+    BASE_METADATA,
+    BASE_METADATA_MAKE_BUNDLE,
+    BASE_METADATA_DATASET_BUNDLE,
+)
+
+
+class BundleManagerMakeBundlesTest(BaseBundleManagerTest):
+    def test_no_bundles(self):
+        self.bundle_manager._make_bundles()
+        self.assertFalse(self.bundle_manager._is_making_bundles())
+
+    def test_restage_stuck_bundle(self):
+        bundle = MakeBundle.construct(
+            targets=[],
+            command='',
+            metadata=BASE_METADATA_MAKE_BUNDLE,
+            owner_id='id1',
+            uuid=generate_uuid(),
+            state=State.MAKING,
+        )
+        self.bundle_manager._model.save_bundle(bundle)
+
+        self.bundle_manager._make_bundles()
+
+        self.assertTrue(self.bundle_manager._is_making_bundles())
+        bundle = self.bundle_manager._model.get_bundle(bundle.uuid)
+        self.assertEqual(bundle.state, State.MAKING)
+
+    def test_bundle_no_dependencies(self):
+        bundle = MakeBundle.construct(
+            targets=[],
+            command='',
+            metadata=BASE_METADATA_MAKE_BUNDLE,
+            owner_id=self.user_id,
+            uuid=generate_uuid(),
+            state=State.STAGED,
+        )
+        self.bundle_manager._model.save_bundle(bundle)
+
+        threads = self.bundle_manager._make_bundles()
+        self.assertTrue(self.bundle_manager._is_making_bundles())
+        for t in threads:
+            t.join()
+        self.assertFalse(self.bundle_manager._is_making_bundles())
+
+        bundle = self.bundle_manager._model.get_bundle(bundle.uuid)
+        self.assertEqual(bundle.state, State.READY)
+
+    def test_single_dependency(self):
+        parent = RunBundle.construct(
```

Comment:
Would be good to refactor these into general utilities for constructing bundles

Reply:
Done